wiki/docs/toolchain.md

```diff
@@ -1,17 +1,47 @@
 Toolchain
 ---
 
-To build Mimiker you will need a toolchain, i.e. compiler, linker, ELF tools and
-debugger. The default option is to choose quite recent _LLVM toolchain_ (check
-version in [tools.mk][6]), i.e. `clang`, `lld`, `llvm` from [apt.llvm.org][7],
-and [gdb-multiarch][8]. However you'll need to reconfigure [launch][9] script to
-use same `gdb-multiarch` binary for each target architecture, look for `gdb`
-section and find `binary` key. 
+To build and run Mimiker you will need a toolchain, i.e. compiler, linker, ELF
+tools emulator and debugger.
 
-You need to install patched version of [QEmu][11] as well. Our version solves
-several issues not patched in the mainstream version – please refer to our list
-of [patches][10]. Prebuild package for Debian x86-64 can be found [here][5].
+## Building tools
+The default option is to choose quite recent _LLVM toolchain_ (check version in
+[tools.mk][6]), i.e. `clang`, `lld`, `llvm` from [apt.llvm.org][7].
 
+List of building tools:
+- make
+- clang
+- llvm
+- lld
+- gperf
+- byacc
+- cpio
+
+## Running tools
+Mimiker is run using [QEmu][11] emulator and can be debugged using
+[gdb-multiarch][8]. There is prepared [launch][9] script to automate running the
+Mimiker (it uses tmux to create windows attached to output and debugger).
+
+NOTE:
+If you plan to run Mimiker on MIPS and RISC-V you need to install patched
+version of QEmu as well. Our version solves several issues not patched in the
+mainstream version – please refer to our list of [patches][10]. Prebuild package
```

Comment:
@cahirwpz I'm not sure if patches to qemu are described with details. Maybe it's worth to send patches to upstream (in spirit of open source)

Reply:
AFAIK https://github.com/cahirwpz/mimiker/blob/master/toolchain/qemu-mimiker/patches/cmdline-set-dts-bootargs.patch is already included into the most recent version of QEmu.

Patch https://github.com/cahirwpz/mimiker/blob/master/toolchain/qemu-mimiker/patches/gdbstub-stop-on-sigxcpu.patch could be made into a valueable contribution to the upstream.

Other ones are MIPS-specific, and since we're planning to phase out this architecture, I would not care much about it.